psydac/pyccel/ast/core.py

```diff
@@ -4605,7 +4605,10 @@ def text(self):
 
     def _sympystr(self, printer):
         sstr = printer.doprint
```

Comment:
According to the Sympy documentation [here](https://docs.sympy.org/latest/modules/printing.html#common-mistakes):

> It’s important to always use `self._print(obj)` to print subcomponents of an expression when customizing a printer

In particular using `printer.doprint` fails when some options are passed to the printer. Here we should write `sstr = printer._print`.

Reply:
done